src/cls/rgw/cls_rgw.cc

```diff
@@ -2391,6 +2570,9 @@ int rgw_dir_suggest_changes(cls_method_context_t hctx,
             return ret;
           }
         }
+        if (header.resharding_in_logrecord()) {
+          return reshard_log_index_operation(hctx, cur_change_key, cur_change.key, &cur_state_bl);
+        }
```

Comment:
can replace this and the `cls_cxx_map_set_val()` call above with `write_entry()`